apps/test/unit/accounts/SchoolInformationTest.jsx

```diff
@@ -0,0 +1,262 @@
+        name: i18n.schoolInformation_updateSchoolInformation(),
+      })
+    );
+
+    await waitFor(() =>
+      expect(
+        screen.getByText(i18n.schoolInformation_updateFailure())
+      ).toBeInTheDocument()
+    );
+    expect(
+      screen.queryByText(i18n.schoolInformation_updateSuccess())
+    ).not.toBeInTheDocument();
+  });
+
+  it('resets alerts when school information changes', async () => {
+    const {rerender} = render(<SchoolInformation {...defaultProps} />);
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: i18n.schoolInformation_updateSchoolInformation(),
+      })
+    );
+
+    // Simulate success alert
+    await waitFor(() =>
+      expect(
+        screen.getByText(i18n.schoolInformation_updateSuccess())
+      ).toBeInTheDocument()
+    );
+
+    // Update school info to trigger useEffect
+    useSchoolInfo.mockReturnValue({
+      ...mockSchoolInfo,
+      schoolName: 'New School',
+    });
+
+    rerender(<SchoolInformation {...defaultProps} />);
+
+    expect(
+      screen.queryByText(i18n.schoolInformation_updateSuccess())
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByText(i18n.schoolInformation_updateFailure())
+    ).not.toBeInTheDocument();
+  });
+
+  it('calls updateSchoolInfo with correct parameters when the button is clicked', async () => {
+    render(<SchoolInformation {...defaultProps} />);
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: i18n.schoolInformation_updateSchoolInformation(),
+      })
+    );
+
+    await waitFor(() => {
+      expect(updateSchoolInfo).toHaveBeenCalledWith({
+        schoolId: '54321',
+        schoolName: 'Test School',
+        schoolZip: '12345',
+        country: 'US',
+      });
+    });
+  });
+
+  it('shows success alert when update is successful', async () => {
+    render(<SchoolInformation {...defaultProps} />);
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: i18n.schoolInformation_updateSchoolInformation(),
+      })
+    );
+
+    // Wait for success alert to appear
+    await waitFor(() => {
+      expect(
+        screen.getByText(i18n.schoolInformation_updateSuccess())
+      ).toBeInTheDocument();
+    });
+    expect(
+      screen.queryByText(i18n.schoolInformation_updateFailure())
+    ).not.toBeInTheDocument();
+  });
+
+  it('shows failure alert when update fails', async () => {
+    updateSchoolInfo.mockRejectedValue(new Error('Update failed'));
+    render(<SchoolInformation {...defaultProps} />);
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: i18n.schoolInformation_updateSchoolInformation(),
+      })
+    );
+
+    // Wait for failure alert to appear
+    await waitFor(() => {
+      expect(
+        screen.getByText(i18n.schoolInformation_updateFailure())
+      ).toBeInTheDocument();
+    });
+    expect(
+      screen.queryByText(i18n.schoolInformation_updateSuccess())
+    ).not.toBeInTheDocument();
+  });
```

Comment:
Are these different from the same-named tests above?

Reply:
oops! nice catch!